sdk/core/azure-core/inc/azure/core/etag.hpp

```diff
@@ -137,10 +138,8 @@ class ETag {
    */
   const std::string& ToString() const
   {
-    if (!m_value.HasValue())
-    {
-      abort();
-    }
+    AZURE_ASSERT_MSG(m_value.HasValue(), "empty eTag");
```

Comment:
```suggestion
    AZURE_ASSERT_MSG(m_value.HasValue(), "empty eTag, m_value.HasValue() should be checked before calling ToString()");
```

Reply:
This assert failure message is for the customer, correct? In that case I wouldn't reference private fields like `m_value` there, because the caller can't write `m_value.HasValue()`.

The assert messages should be written in a way that is relevant and actionable to the end user.